### ISSUE TITLE
Add more basic assertions for Local(Date|Time)

### ISSUE THREAD

**Issue by onacit** [assertj/assertj-core#2541]
#### Summary

It would be nice if `AbstractLocal(Date|Time)Assert` has basic assertions for field values of `actual`.

LocalDate
- [x] `hasYear` - 9dc086d82aa58a755f36cf88449b0ca74288bb63
- [x] `hasMonth` / `hasMonthValue` - 40c979c253571743af8d68f18921a061e10cfcec
- [x] `hasDayOfMonth` - 253194d00395e1568e1be752b0075c2033672f97

LocalTime
- [x] `hasHour` - 8c54e417b697314e55170668bd96c236fa3c96d6
- [x] `hasMinute` - ec1fcaefb767d0ed18a7e6ea48779c78635d1399
- [x] `hasSecond` - 6a4e54cbad5198246ae41a21d8432c503b5c70ce
- [x] `hasNano` - b756459

LocalDateTime
- [x] `hasYear` - 710177b
- [x] `hasMonth` / `hasMonthValue`  - d0c87895c2a052473e8c4a7800fa7fee0a20dccb
- [x] `hasDayOfMonth` - 710177b
- [x] `hasHour` - 7cbe45c42f0a31bc5993a8352cf5e0d4feda0fae
- [ ] `hasMinute` - @Leo0506
- [x] `hasSecond` - 99e4f652cecc6f4492a0aec7bcfc7e97518a032a
- [x] `hasNano` -  76012bb

#### Examples

```java
  final LocalDate actual = LocalDate.now();
  assertThat(actual).hasYear(actual.getYear());
  assertThat(actual).hasMonth(actual.getMonth());
  assertThat(actual).hasMonthValue(actual.getMonthValue());
  assertThat(actual).hasDayOfMonth(actual.getDayOfMonth());

  final LocalTime actual = LocalTime.now();
  assertThat(actual).hasHour(actual.getHour());
  assertThat(actual).hasMinute(actual.getMinute());
  assertThat(actual).hasSecond(actual.getSecond());
  assertThat(actual).hasNano(actual.getNano());
```

**Comment by jessicant** [comment]
HI. I would like to work on this. Please assign me.

**Comment by joel-costigliola** [comment]
@jessicant thanks! 
Can you start with `hasYear` and raise a PR?
The idea is that we review it so that the other assertions are going to be easy to do ;-)
You can also only do `hasYear` if you feel that issue is too much work.

**Comment by jessicant** [comment]
Hi, my friend @nith2001 will comment on this issue soon. Will you please make him an assignee as well? We would like to work on it together :)

**Comment by nith2001** [comment]
Hello! Verifying what Jess said!

**Comment by jessicant** [comment]
@joel-costigliola was there a reason we were unassigned? We would still like to work on this issue, but we can pick another if need be

**Comment by scordio** [comment]
@jessicant I assume @joel-costigliola unassigned you only because `hasYear` was merged. Would you mind announcing here what you will choose next so that we can avoid overlap with other contributors?

**Comment by jessicant** [comment]
@scordio thank you for assigning us. We will work on `LocalDate hasMonth/hasMonthValue` next.

**Comment by joel-costigliola** [comment]
thanks @jessicant, this issue being big I split it into tasks so that multiple contributors can work on it, they should as you did indicate which assertions they want to work on.
the integrated commit can be used as a reference for implementing the other assertions: https://github.com/assertj/assertj-core/commit/9dc086d82aa58a755f36cf88449b0ca74288bb63is

**Comment by nith2001** [comment]
Dope. Jessica and I will be working on it fully this Tuesday and we should be able to get a good chunk done then.

**Comment by jessicant** [comment]
Hi @onacit, the example is a little ambiguous. For `hasMonth()` do you want it to work with `int` values and/or `Month` objects? We can write both, but wanted to check before we start adding features ;)

**Comment by joel-costigliola** [comment]
@jessicant this why @onacit has suggested having two assertions for month that matches the LocalDate API
- https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html#getMonthValue--
- https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html#getMonth--
Use `int` for `hasMonth` and `Month` for `hasMonthValue`.

@onacit correct me if I'm wrong ;-)

**Comment by nith2001** [comment]
Awesome, I have it set up that way where one function takes Month enum and the other takes the month value.

**Comment by Leo0506** [comment]
Hello, I'm interested in this issue. Could you assign me the functions hasHour and hasMinute in LocalDateTime?

**Comment by maxwell142857** [comment]
Hi, actually SariaCxs is my friend and I want to work on  the assertion of hassecond and hasNano in LocalTime, can I work on these two sub-tasks?

**Comment by nith2001** [comment]
We'll take `hasDayOfMonth()` to finish off the LocalDate asserts.

**Comment by scordio** [comment]
Topics assigned as requested. Thank you all for your interest!

**Comment by YeeTone** [comment]
I am interested in hasSecond of Local date time. Could you assign me to this feature?

**Comment by YeeTone** [comment]
I am interested in hasSecond of LocalDateTime. Could you assign me to this feature?

**Comment by Zhou-Yicheng** [comment]
Could you assign LocalDateTime's `hasYear`, `hasMonth / hasMonthValue` and `hasDayOfMonth` to me?

**Comment by YeeTone** [comment]
Actually, @Zhou-Yicheng is my friend. I am interested in `hasSecond` of `LocalDateTime`. Could you assign me to this feature? @scordio

**Comment by joel-costigliola** [comment]
@Zhou-Yicheng I have assigned you `hasMonth / hasMonthValue` that should be enough for a start, please look at https://github.com/assertj/assertj-core/pull/2566 for reference when that PR is merged.

**Comment by jessicant** [comment]
May I take 'hasNano' of 'LocalDateTime' next please?

**Comment by MatthewLENG2002** [comment]
Hello, I'm interested in this issue, could you assign LocalDateTime's `hasYear`  and `hasDayOfMonth` to me?

**Comment by MatthewLENG2002** [comment]
@scordio @joel-costigliola Hello, I'm interested in this issue, could you assign LocalDateTime's `hasYear` and `hasDayOfMonth` to me?

**Comment by scordio** [comment]
Done, @MatthewLENG2002. Thanks for your help!

**Comment by joel-costigliola** [comment]
@Leo0506 are you tackling the last assertion of this issue? We are almost done!

**Comment by Leo0506** [comment]
> 

Oh yeah. Sorry to be a little bit late.
